goto/tests/test_end_to_end.sh

```diff
@@ -217,26 +217,57 @@ function test_06_goto_add {
     _projectfile_should_contain "test_escaped_ampersand_in_url"
 }
 
-function test_07_goto {
-    magicword="test_goto"
-    testcdpath="/tmp"
-    nonexisting_magicword="IDoNotExist"
-    uri="http://example.com"
+function test_07_goto_many {
+    declare -a magicwords
+    declare -a uris
+    magicwords=(
+        "red"
+        "blue"
+        "green"
+        "yellow"
+        "orange"
+    )
+    uris=(
+        "\"http://red.com\""
+        "\"mailto:jon@gmail.com\""
+        "\"spotify:green.com\""
+        "\"sftp:hallo@yellow.com\""
+        "\"goto+subl://orange.com\""
+    )
+
+    for i in "${!magicwords[@]}"; do
+        _cmd_should_succeed "goto add ${magicwords[$i]} ${uris[$i]}";
+    done
 
-    _cmd_should_succeed "goto add $magicword $uri"
-    _cmd_should_fail "goto $nonexisting_magicword"
-    _failing_cmd_should_give_human_message "goto $nonexisting_magicword"
-    _failing_cmd_should_not_print_ah_hoy_twice "goto $nonexisting_magicword"
+    for i in "${!magicwords[@]}"; do
+        _projectfile_should_contain "${magicwords[$i]}";
+    done
+
+    for i in "${!magicwords[@]}"; do
+        _cmd_should_succeed "goto ${magicwords[$i]}";
```

Comment:
Wow - it is actually a little worrying that these commands succeed now, as `goto+subl` is not a registered scheme on any computer at the time of this writing.

This is an old bug or misbehaving feature of goto if you want to call it that

Reply:
It might be considered outside the scope of this PR to fix the exit status of goto <magicwords> that should fail